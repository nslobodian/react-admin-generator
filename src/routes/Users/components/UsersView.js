import React, { Component } from 'react'
import PropTypes from 'prop-types'
import UsersTable from './UsersTable'
import PaginationWrapper from 'components/PaginationWrapper'
import Spinner from 'components/Spinner'

class UsersView extends Component {
  componentDidMount () {
    this.props.initChangeFilter(this.props.location.query)
  }

  componentWillUnmount () {
    this.props.resetFilters()
  }

  render () {
    const { users, handleChangeFilter, getUsersPagination, goToUserDetails, getUsersLoading, pagination } = this.props
    return (
      <div>
        <div className='panel panel-default'>
          <div className='panel-heading'>Members</div>
          {getUsersLoading && <Spinner />}
          <div>
            <PaginationWrapper {...pagination}
              label='members'
              handleChange={getUsersPagination}
            >
              {
                users &&
                <UsersTable users={users}
                  handleChangeFilter={handleChangeFilter}
                  goToUserDetails={goToUserDetails} />
              }
            </PaginationWrapper>
          </div>
        </div>
      </div>
    )
  }
}

UsersView.propTypes = {
  handleChangeFilter: PropTypes.func,
  users: PropTypes.array,
}

export default UsersView
