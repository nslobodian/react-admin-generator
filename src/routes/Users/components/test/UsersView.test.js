import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import UsersView from '../UsersView'

const props = {
  handleChangeFilter: jest.fn(),
  getUsersPagination: jest.fn(),
  goToUserDetails: jest.fn(),
  getUsersLoading: jest.fn(),
  pagination: {
    'page': 1,
    'pages': 1,
    'perPage': 10,
    'totalCount': 5,
  },
  users: [{
    username: 'Username',
    url: 'google.com',
    lastName: 'LastName',
    name: 'name',
    email: 'email@dot.com',
    userId: 1234,
  }],
  handleChangeFilter: jest.fn(),
  goToUserDetails: jest.fn(),
  resetFilters: jest.fn(),
  getUsersAttempt: jest.fn(),
}

describe('(Route) UsersView', () => {
  it('Should render component properly and unmount', () => {
    const wrapper = shallow(
      <UsersView
        {...props}
      />)
    expect(toJson(wrapper)).toMatchSnapshot()
    wrapper.unmount()
  }),
  it('Should render component properly and unmount', () => {
    const wrapper = shallow(
      <UsersView
        {...props}
      />)
    wrapper.instance().componentDidMount()
  })
})
