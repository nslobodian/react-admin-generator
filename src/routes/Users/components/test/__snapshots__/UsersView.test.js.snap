// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Route) UsersView Should render component properly and unmount 1`] = `
<div>
  <div
    className="panel panel-default"
  >
    <div
      className="panel-heading"
    >
      Users
    </div>
    <Component />
    <div>
      <PaginationWrapper
        handleChange={[Function]}
        label="users"
        page={1}
        pages={1}
        perPage={10}
        totalCount={5}
      >
        <ReduxForm
          goToUserDetails={[Function]}
          handleChangeFilter={[Function]}
          users={
            Array [
              Object {
                "email": "email@dot.com",
                "lastName": "LastName",
                "name": "name",
                "url": "google.com",
                "userId": 1234,
                "username": "Username",
              },
            ]
          }
        />
      </PaginationWrapper>
    </div>
  </div>
</div>
`;
